{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import io\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "ICGC_SAMPLE_ID = 'SA320818' #Sample we are reading\n",
    "df = pd.read_csv('E:/Sequencing/simple_somatic_mutation/samples/' + ICGC_SAMPLE_ID + '.tsv', \\\n",
    "                 sep = '\\t', low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "pd.set_option('display.max_columns', 5000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_from_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "      <th>total_read_count</th>\n",
       "      <th>mutant_allele_read_count</th>\n",
       "      <th>consequence_type</th>\n",
       "      <th>gene_affected</th>\n",
       "      <th>transcript_affected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SA320818</td>\n",
       "      <td>3</td>\n",
       "      <td>147424632</td>\n",
       "      <td>147424632</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>41.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SA320818</td>\n",
       "      <td>3</td>\n",
       "      <td>147424632</td>\n",
       "      <td>147424632</td>\n",
       "      <td>1</td>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>35.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SA320818</td>\n",
       "      <td>5</td>\n",
       "      <td>75054899</td>\n",
       "      <td>75054899</td>\n",
       "      <td>1</td>\n",
       "      <td>single base substitution</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>25.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SA320818</td>\n",
       "      <td>15</td>\n",
       "      <td>60419020</td>\n",
       "      <td>60419020</td>\n",
       "      <td>1</td>\n",
       "      <td>single base substitution</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>42.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SA320818</td>\n",
       "      <td>17</td>\n",
       "      <td>71121414</td>\n",
       "      <td>71121414</td>\n",
       "      <td>1</td>\n",
       "      <td>insertion of &lt;=200bp</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>30.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_sample_id chromosome  chromosome_start  chromosome_end  \\\n",
       "0       SA320818          3         147424632       147424632   \n",
       "1       SA320818          3         147424632       147424632   \n",
       "2       SA320818          5          75054899        75054899   \n",
       "3       SA320818         15          60419020        60419020   \n",
       "4       SA320818         17          71121414        71121414   \n",
       "\n",
       "   chromosome_strand             mutation_type reference_genome_allele  \\\n",
       "0                  1       deletion of <=200bp                       T   \n",
       "1                  1       deletion of <=200bp                       T   \n",
       "2                  1  single base substitution                       C   \n",
       "3                  1  single base substitution                       C   \n",
       "4                  1      insertion of <=200bp                       -   \n",
       "\n",
       "  mutated_from_allele mutated_to_allele  total_read_count  \\\n",
       "0                   T                 -              41.0   \n",
       "1                   T                 -              35.0   \n",
       "2                   C                 T              25.0   \n",
       "3                   C                 T              42.0   \n",
       "4                   -                 A              30.0   \n",
       "\n",
       "   mutant_allele_read_count   consequence_type gene_affected  \\\n",
       "0                      13.0  intergenic_region           NaN   \n",
       "1                      12.0  intergenic_region           NaN   \n",
       "2                      18.0  intergenic_region           NaN   \n",
       "3                       8.0  intergenic_region           NaN   \n",
       "4                      12.0  intergenic_region           NaN   \n",
       "\n",
       "  transcript_affected  \n",
       "0                 NaN  \n",
       "1                 NaN  \n",
       "2                 NaN  \n",
       "3                 NaN  \n",
       "4                 NaN  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#5,9-12,14-17,20-21,26,29-30"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
